bcgov/mds
# [MDS-4642] Major Mine Application - Core - Add files to permit package

## Objective 

[MDS-4642](https://bcmines.atlassian.net/browse/MDS-4642), [MDS-4641](https://bcmines.atlassian.net/browse/MDS-4641) 

_Major Changes_

- Created models, resources, responses, tests for new project_decision_package related entities
- Integrated existing frontend view with new API
- Added new static content for projectDecisionPackageStatusCodes
- Added new permission to API: "core_edit_project_decision_packages"
- Created and integrated two new modals:
    - Upload document for decision package
    - Update decision package files
- Added dynamic alert box with ability to change status

_How to Test_

- New: Navigate to "Decision Package" tab from Project Overview: 
![image](https://user-images.githubusercontent.com/13891487/188232252-6d4b9cda-f171-4933-9b2d-072c8af55ea0.png)
    - Verify status alert at top of page is "Not Started"
    - Verify "Edit Package" and "Add Documents" buttons are disabled
    - Verify after changing status to "In Progress" that the "Add Documents" buttons is enabled and alert box updates: 
![image](https://user-images.githubusercontent.com/13891487/188232385-770913ba-a1e3-4226-8e11-eac13bbd1921.png)
    - Verify you are able to add documents both flagged as "Decision Package" or not(Additional Government Documents section):
 ![image](https://user-images.githubusercontent.com/13891487/188233103-de7295aa-9968-4f7c-902b-84c5ab3066cd.png)
    - Verify you are able to add documents to the "Internal Ministry Documentation" section(button at bottom of page):  
![image](https://user-images.githubusercontent.com/13891487/188233261-1ea9a32b-400f-4d9d-82a3-a8135e04c0eb.png)
- Existing: While on the Decision Package tab: 
![image](https://user-images.githubusercontent.com/13891487/188233407-e07ef1b5-52b8-4d8c-84cf-1db6434b9371.png)
    - Verify you can click "Edit Package" and a separate modal shows up with documents in it(these documents will be either type "DCP" or "ADG", documents that are "DCP" will be checked off
      - Verify the right files are checked off initially
      - Verify the "Select All" functionality on the top left of the table works
      - Verify changing the document types and clicking "Update" works as expected
